opennms-webapp/src/main/java/org/opennms/web/element/NetworkElementFactory.java

```diff
@@ -1009,4 +1020,50 @@ public boolean nodeExistsInRequisition(final String foreignSource, final String
     public Collection<OnmsOutage> currentOutagesForServiceFromPerspectivePoller(OnmsMonitoredService service){
         return m_outageDao.currentOutagesForServiceFromPerspectivePoller(service);
     }
+
+    @Override
+    public List<OnmsApplication> getAllApplications() {
+        return m_applicatioDao.findMatching(new CriteriaBuilder(OnmsApplication.class).orderBy("name").toCriteria());
+    }
+
+    @Override
+    public ApplicationStatus getApplicationStatus(final OnmsApplication onmsApplication, final long start, final long end) {
+        return ApplicationStatusUtil.buildApplicationStatus(
+                onmsApplication,
+                m_outageDao.getStatusChangesForApplicationIdBetween(new Date(start), new Date(end), onmsApplication.getId()),
+                start,
+                end
+        );
+    }
+
+    @Override
+    public Map<OnmsMonitoredService, Map<String, Double>> getApplicationServiceStatus(final OnmsApplication onmsApplication, final long start, final long end) {
+        final Map<String, OnmsMonitoredService> serviceMap = new TreeMap<>();
```

Comment:
The key of the map is never used. Can `onmsApplication.getMonitoredServices()` be used directly in the loop below?

Reply:
done.